packages/gatsby-cli/src/reporter/loggers/ink/components/messages.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react"
+import { Box, Color, ColorProps } from "ink"
+
+import { ActivityLogLevels, LogLevels } from "../../../constants"
+
+const ColorSwitcher: React.FC<ColorProps> = ({ children, ...props }) => (
+  <Color {...props}>{children}</Color>
+)
+
+const createLabel = (text: string, color: string): React.FC<ColorProps> => (
+  ...props
+): JSX.Element => (
+  <ColorSwitcher {...{ [color]: true, ...props }}>{text}</ColorSwitcher>
+)
```

Comment:
I couldn't narrow the type of `color` into possible values which are ​​determined depending on `ColorProps` so I defined it as `string`. 
Ideally it should be like `"black" | "red" | "green" | "yellow" | ...`.